java-checks/src/main/java/org/sonar/java/checks/CollectionImplementationReferencedCheck.java

```diff
@@ -109,35 +243,54 @@ public void visitVariable(VariableTree tree) {
   @Override
   public void visitMethod(MethodTree tree) {
     super.visitMethod(tree);
-    if (isPublic(tree.modifiers()) && Boolean.FALSE.equals(tree.isOverriding())) {
-      checkIfAllowed(tree.returnType(), "The return type of this method");
-      for (VariableTree variableTree : tree.parameters()) {
-        checkIfAllowed(variableTree.type(), String.format("The type of \"%s\"", variableTree.simpleName()));
-      }
+    if (!isPublic(tree.modifiers()) || !Boolean.FALSE.equals(tree.isOverriding())) {
+      return;
+    }
+
+    checkIfAllowed(tree.returnType(), "The return type of this method");
+    var candidateParameters = tree.parameters().stream()
```

Comment:
I think it would be better to have the type of the resulting collection for clarity
```suggestion
    List<VariableTree> candidateParameters = tree.parameters().stream()
```

Reply:
Fixed.